tests/downloaders/test_thesesfr.py

```diff
@@ -0,0 +1,62 @@
+import pytest
+
+from papis.downloaders.thesesfr import Downloader
+
+
+def test_match():
+    valid_urls = [
+        "https://www.theses.fr/2014TOU30305",
+        "2014TOU30305",
+        "https://www.theses.fr/2014TOU30305.bib/?asdf=2",
+    ]
+    invalid_urls = ["http://google.com", "2014TOU", "ASDF"]
+    for url in valid_urls:
+        assert isinstance(Downloader.match(url), Downloader)
+
+    for url in invalid_urls:
+        assert Downloader.match(url) is None
+
+
+@pytest.mark.parametrize(
+    ("query", "expected"),
+    [
+        ("https://www.theses.fr/2014TOU30305", "2014TOU30305"),
+        ("https://www.theses.fr/2014TOU30305.bib/?asdf=2", "2014TOU30305"),
+        ("2014TOU30305", "2014TOU30305"),
+        ("2014TOU", None),
+    ],
+)
+def test_get_identifier(query: str, expected: str):
```

Comment:
The tests should all have a `tmp_config: TemporaryConfiguration` fixture added so that they work in a clean environment.

Reply:
I'll add that!